Refactor/ Migrate Koin To Hilt Dagger

AhmadKharfan opened the pull request:
### Replace Koin with Hilt + Dagger; move nav args to SavedStateHandle

#### Highlights
- **DI**
  - Drop Koin.
  - Add Hilt to Android/framework modules.
  - Use plain Dagger 2 + KSP in pure‑Kotlin domain; expose it to Hilt via one bridge.

- **Navigation / ViewModels**
  - Single‑source `Destination` objects; arguments read from `SavedStateHandle`.
  - All VMs converted to `@HiltViewModel` + `@Inject` constructors.

- **Gradle**
  - Add `dagger`, `dagger‑compiler`, `ksp`.
  - Remove Koin artifacts.

AhmadKharfan committed:
add hilt dependencies for dependency injection

AhmadKharfan committed:
migrate details koin modules to hilt

AhmadKharfan committed:
use hilt for view models and update dispatcher

AhmadKharfan committed:
update screens to use hilt view models

AhmadKharfan committed:
add local search data source module

AhmadKharfan committed:
use in search feature and app hilt instead of koin

AhmadKharfan committed:
use hilt view model in search ui

AhmadKharfan committed:
integrate novix app with feature apis

AhmadKharfan committed:
remove koin and use hilt for dependency injection in details screens

AhmadKharfan committed:
inject search navigator via hilt constractor

AhmadKharfan committed:
remove koin and update hilt dependencies

AhmadKharfan committed:
ensure hilt modules use singleton scope

AhmadKharfan committed:
use inject in data sources

AhmadKharfan committed:
add inject to repository constructors

AhmadKharfan committed:
remove playlist and profile features from main screen

AhmadKharfan committed:
resolve hilt integration issues

AhmadKharfan committed:
jakarta inject to javax inject

AhmadKharfan committed:
standardize di module parameter names

AhmadKharfan committed:
add dagger dependencies to vod domain module

AhmadKharfan committed:
remove redundant hilt modules

AhmadKharfan committed:
add dagger inject to use case constructors

AhmadKharfan committed:
rename series id to media in reviews screen route and fix convert from string to ui state

AhmadKharfan committed:
convert remote details data source module from an object to an abstract class with binds methods for data source provision

AhmadKharfan committed:
add inject constructor to remote data sources and remove unused imports

ZainabAlnajjar commented on feature/home/presentation/src/main/java/com/sanaa/presentation/screen/celebritiesScreen/CelebritiesViewModel.kt:
```diff
@@ -2,12 +2,17 @@ package com.sanaa.presentation.screen.celebritiesScreen
 ) : BaseViewModel<CelebritiesScreenUiState, CelebritiesScreenEffects>(
-    CelebritiesScreenUiState()
+    initialState = CelebritiesScreenUiState(),
+    defaultDispatcher = Dispatchers.IO
```
please take the dev changes for the dispatcher when you merge this part, so the tests wont fail

AhmadKharfan committed:
merge develop and resolve conflicts

AhmadKharfan committed:
resolve conflicts in api and logging modules

AhmadKharfan committed:
remove unnecessary firebase injection

AhmadKharfan committed:
add coroutine dispatcher provider

AhmadKharfan committed:
inject id dispatcher in view models

AhmadKharfan committed:
use saved state handle in view model tests

AhmadKharfan committed:
Merge branch 'development' into refactor/migrate-koin-to-hilt-dagger

AhmadKharfan committed:
remove redundant tests from series view model test

AhmadKharfan committed:
improve test coverage report

AhmadKharfan committed:
merge develop and resolve conflicts

AhmadKharfan committed:
solve di problems

AhmadKharfan committed:
solve di problems

AhmadKharfan committed:
enable hilt injection for stringProviderImpl and main activity

AhmadKharfan committed:
add authentication dependencies and hilt

AhmadKharfan committed:
Refactor: Use Hilt for dependency injection

This commit migrates the following classes from Koin to Hilt for dependency injection:

- `PreferencesManagerImpl`
- `LoginViewModel`
- `WelcomeViewModel`
- `LoginUseCase`
- `AuthenticationRepositoryImpl`

Additionally, it updates `WelcomeScreen` and `LoginScreen` to use `hiltViewModel()` for ViewModel instantiation.

AhmadKharfan committed:
inject home feature api into auth nav host

AhmadKharfan committed:
fix tests and update viewModel initialization

AhmadKharfan committed:
Merge remote-tracking branch 'origin/development' into refactor/migrate-koin-to-hilt-dagger

AhmadKharfan committed:
update tests for AuthenticationApiImpl to reflect the new dependency on HomeFeatureApi

AhmadKharfan committed:
remove custom dispatcher from welcome view model test
